flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/common/TieredStorageConfiguration.java

```diff
@@ -18,58 +18,408 @@
+     * @return the buffer number.
+     */
+    public int getAccumulatorExclusiveBuffers() {
+        return accumulatorExclusiveBuffers;
+    }
+
+    /**
+     * Get the segment size of memory tier.
+     *
+     * @return segment size.
+     */
+    public int getMemoryTierNumBytesPerSegment() {
+        return memoryTierNumBytesPerSegment;
+    }
+
+    /**
+     * Get the segment size of disk tier.
+     *
+     * @return segment size.
+     */
+    public int getDiskTierNumBytesPerSegment() {
+        return diskTierNumBytesPerSegment;
+    }
+
+    /**
+     * Get the segment size of remote tier.
+     *
+     * @return segment size.
+     */
+    public int getRemoteTierNumBytesPerSegment() {
+        return remoteTierNumBytesPerSegment;
+    }
+
+    /**
+     * When the number of buffers that have been requested exceeds this threshold, trigger the
+     * flushing operation in each {@link TierProducerAgent}.
+     *
+     * @return flush ratio.
+     */
+    public float getNumBuffersTriggerFlushRatio() {
+        return numBuffersTriggerFlushRatio;
+    }
+
+    /**
+     * The number of buffers to read ahead at most for each subpartition in {@link DiskIOScheduler},
+     * which can be used to prevent other consumers from starving.
+     *
+     * @return buffer number.
+     */
+    public int getDiskIOSchedulerMaxBuffersReadAhead() {
+        return diskIOSchedulerMaxBuffersReadAhead;
+    }
+
+    /**
+     * Maximum time to wait when requesting read buffers from the buffer pool before throwing an
+     * exception in {@link DiskIOScheduler}.
+     *
+     * @return timeout duration.
+     */
+    public Duration getDiskIOSchedulerBufferRequestTimeout() {
+        return diskIOSchedulerRequestTimeout;
+    }
+
+    /**
+     * Minimum reserved disk space fraction in disk tier.
+     *
+     * @return the fraction.
+     */
+    public float getMinReserveSpaceFraction() {
+        return minReserveSpaceFraction;
+    }
+
+    /**
+     * Get the total exclusive buffer number.
+     *
+     * @return the total exclusive buffer number.
+     */
+    public int getTotalExclusiveBufferNum() {
+        return accumulatorExclusiveBuffers
+                + memoryTierExclusiveBuffers
+                + diskTierExclusiveBuffers
+                + (remoteStorageBasePath == null ? 0 : remoteTierExclusiveBuffers);
+    }
+
+    /**
+     * Get exclusive buffer number of each tier.
+     *
+     * @return buffer number of each tier.
+     */
+    public List<Integer> getEachTierExclusiveBufferNum() {
+        List<Integer> tierExclusiveBuffers = new ArrayList<>();
+        tierExclusiveBuffers.add(memoryTierExclusiveBuffers);
+        tierExclusiveBuffers.add(diskTierExclusiveBuffers);
+        if (remoteStorageBasePath != null) {
+            tierExclusiveBuffers.add(remoteTierExclusiveBuffers);
+        }
+        return tierExclusiveBuffers;
+    }
+
+    public List<TierFactory> getTierFactories() {
+        return tierFactories;
+    }
```

Comment:
These are not consistent. `tierFactories` may or may not be the default factories, while the number of buffers for each tier is hard coded. I think at least we should add a `createDefaultTierExclusiveBufferNums` in the builder, so that we use the default list of buffer numbers only if we use the default factories. A better way might be have a `createDefultTierFactoriesAndMemorySpecs` that decide the hard coded factories and memory parameters at once.